src/test/fuzz/util.h

```diff
@@ -534,4 +544,118 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSock : public Sock
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSock(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ~FuzzedSock() override
+    {
+    }
+
+    SOCKET Get() const override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    SOCKET Release() override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    void Reset() override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    ssize_t Send(const void* data, size_t len, int flags) const override
+    {
+        constexpr std::array send_errnos{
```

Comment:
I think we want to mimic a conforming OS `send()` because our code is not prepared to handle a misbehaving `send()`, e.g.
* one that returns `-1` and does not set `errno` or
* returns `-1` and sets errno to `2379392` or 
* returns `-9287` or
* returns a number bigger than `len`.